src/compiler/marks.ts

```diff
@@ -357,8 +357,53 @@ export namespace bar {
   }
 
   export function labels(model: Model) {
-    // TODO(#64): fill this method
-    return undefined;
+    // TODO(#64):
+    // √ horizontal support
+    // - vertical support
+    // - color support
+    // - container height/stretching consideration
+    // - consider defaults for offset
+
+    var l: any = {};
+    const orient = model.marksConfig('orient');
+
+    l.align = { value: (orient !== 'horizontal') ? "center" : "left" };
+
+    l.text = { field: model.has(LABEL) ? model.field(LABEL) : ((orient !== 'horizontal') ? model.field(Y) : model.field(X)) };
```

Comment:
If you look at `model.has()` in `Model.ts` -- `has` actually check existence of the field.   So we can be sure about that. 

(I start wondering if we should rename `has` to `hasField`..)